addons/web/static/tests/views/kanban/kanban_view.test.js

```diff
@@ -4495,6 +4495,87 @@ test.tags("desktop")("can drag and drop a record from one column to the next", a
     expect(["resequence"]).toVerifySteps();
 });
 
+test.tags("desktop")(
+    "user without permission cannot drag and drop a column thus sequence remains unchanged on drag and drop attempt",
+    async () => {
+        expect.errors(1);
+
+        onRpc("/web/dataset/resequence", () => {
+            throw makeServerError({ message: "No Permission" }); // Simulate user without permission
+        });
+
+        await mountView({
+            type: "kanban",
+            resModel: "partner",
+            arch: `
+                <kanban>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div class="oe_kanban_global_click">
+                                <field name="foo"/>
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            groupBy: ["product_id"],
+        });
+
+        expect(queryAllTexts(".o_column_title")).toEqual(["hello", "xmo"]);
+
+        const groups = queryAll(".o_column_title");
+        await contains(groups[0]).dragAndDrop(groups[1]);
+
+        expect(queryAllTexts(".o_column_title")).toEqual(["hello", "xmo"]);
+
+        expect(["No Permission"]).toVerifyErrors();
+    }
+);
+
+test.tags("desktop")(
+    "user without permission cannot drag and drop a record thus sequence remains unchanged on drag and drop attempt",
+    async () => {
+        expect.errors(1);
+
+        onRpc("partner", "web_save", () => {
+            throw makeServerError({ message: "No Permission" }); // Simulate user without permission
+        });
+
+        await mountView({
+            type: "kanban",
+            resModel: "partner",
+            arch: `
+                <kanban>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div class="oe_kanban_global_click">
+                                <field name="foo"/>
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            groupBy: ["product_id"],
+        });
+
+        expect(".o_kanban_record:first").toHaveText("yop", {
+            message: "Checking the initial state of the view",
+        });
+
+        await contains(".o_kanban_record").dragAndDrop(".o_kanban_group:nth-child(2)");
+
+        expect(".o_kanban_record:first").toHaveText("yop", {
+            message: "Do not let the user d&d the record without permission",
+        });
+
+        await contains(".o_kanban_record").dragAndDrop(".o_kanban_record:nth-child(3)");
```

Comment:
Because the RPC call for this d&d is different and we are not throwing error on it in the test case. My motive for this d&d is to just check that the record does not become static after a failed d&d.